core/src/test/java/org/apache/calcite/test/RexProgramBuilderBase.java

```diff
@@ -433,7 +436,7 @@ protected RexNode literal(String value) {
    * @param arg argument index (0-based)
    * @return input ref with given type and index
    */
-  protected RexNode input(RelDataType type, int arg) {
+  protected RexInputRef input(RelDataType type, int arg) {
```

Comment:
Did you change this to make `item(...)` happy?
I guess it should be the other way around. `item(...)` should be adjusted to support `RexNode` argument.

Reply:
yes, agree)